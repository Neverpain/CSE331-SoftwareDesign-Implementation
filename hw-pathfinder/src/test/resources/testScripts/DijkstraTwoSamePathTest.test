# Tests that Dijkstra's algorithm doesn't have defined behavior for outputting paths
# when there are two minimum-cost paths

CreateGraph g

AddNode g A
AddNode g B
AddNode g C
AddNode g D

AddEdge g A B 1.0
AddEdge g B D 3.0
AddEdge g A C 2.0
AddEdge g C D 2.0

FindPath g A D