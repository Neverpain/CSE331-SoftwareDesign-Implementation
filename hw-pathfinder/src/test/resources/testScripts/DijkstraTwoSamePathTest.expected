# Tests that Dijkstra's algorithm doesn't have defined behavior for outputting paths
# when there are two minimum-cost paths

created graph g

added node A to g
added node B to g
added node C to g
added node D to g

added edge 1.000 from A to B in g
added edge 3.000 from B to D in g
added edge 2.000 from A to C in g
added edge 2.000 from C to D in g

path from A to D:
A to B with weight 1.000
B to D with weight 3.000
total cost: 4.000